Problems with tests on Fedora

Hi,
I'm updating the Fedora package, but I'm not able to make the tests work. I'm not sure if it's a problem with how Fedora builds Python or there is something else going on.
I would say that the majority of tests works properly, but the SHA3 ones fail with an error like the following (the type it expects does change on the various Python versions):

```
======================================================================
FAIL: test_run_mypy (tests.test_mypy.MypyRunnerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/rsa-4.6/tests/test_mypy.py", line 27, in test_run_mypy
    self.fail('\n'.join(['Mypy errors:'] + messages))
AssertionError: Mypy errors:
rsa/pkcs1.py:65: error: Dict entry 0 has incompatible type "str": "Type[_Hash]"; expected "str": "Callable[[Union[bytes, memoryview]], _Hash]"
rsa/pkcs1.py:66: error: Dict entry 1 has incompatible type "str": "Type[_Hash]"; expected "str": "Callable[[Union[bytes, memoryview]], _Hash]"
rsa/pkcs1.py:67: error: Dict entry 2 has incompatible type "str": "Type[_Hash]"; expected "str": "Callable[[Union[bytes, memoryview]], _Hash]"
Found 3 errors in 1 file (checked 30 source files)

Mypy failed with status 1

----------------------------------------------------------------------
Ran 94 tests in 7.514s

FAILED (failures=1)
Test failed: <unittest.runner.TextTestResult run=94 errors=0 failures=1>
	Message:   b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01'
	Encrypted: b'1\x8c\x82\xb7)\x81w\xe7$\x0bB\x92I\xae\xee\x9dT\xd1\xec\x1d\x0f\xb9/\xdb\xa3\xf1w+%o\xb41'
	Decrypted: b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01'
	Message:   b'Euro=\xe2\x82\xac ABCDEFGHIJKLMNOPQRSTUVWXYZ'
	Encrypted: b'V^\x81\x02\xc5\xc72\x93\x9c\x13\xa1\xfc\xb7!\xf0\x93\x15oE\xdek\x11\xc8a\xaa\xa9C\xa5\xd2\x1a\xcc\x9b\xc8\x11\xd5\x80\xf2\xa8\xd75\x17\xa0\x959\x9f\xa7\xfc\xe9'
	Decrypted: b'Euro=\xe2\x82\xac ABCDEFGHIJKLMNOPQRSTUVWXYZ'
error: Test failed: <unittest.runner.TextTestResult run=94 errors=0 failures=1>
```

This happens with:
* Fedora 31 (Python 3.7.7) [full logs](https://fale.fedorapeople.org/python-rsa/fedora31.log)
* Fedora 32 (Python 3.8.3) [full logs](https://fale.fedorapeople.org/python-rsa/fedora32.log)
* Fedora 33 (Python 3.9.0~b1) [full logs](https://fale.fedorapeople.org/python-rsa/fedora33.log)

Do you have any idea on the possible cause and work around?

Thanks a lot!

--- a/.coveragerc
+++ b/.coveragerc
@@ -1,5 +1,9 @@
 [report]
 # Regexes for lines to exclude from consideration
 exclude_lines =
+    # Re-enable the standard pragma
+    pragma: no cover
     # Don't complain if non-runnable code isn't run
     if __name__ == .__main__.:
+    # TYPE_CHECKING is False at runtime
+    if typing.TYPE_CHECKING:

--- a/rsa/pkcs1.py
+++ b/rsa/pkcs1.py
@@ -34,6 +34,11 @@ from hmac import compare_digest
 
 from . import common, transform, core, key
 
+if typing.TYPE_CHECKING:
+    HashType = hashlib._Hash
+else:
+    HashType = typing.Any
+
 # ASN.1 codes that describe the hash algorithm used.
 HASH_ASN1 = {
     'MD5': b'\x30\x20\x30\x0c\x06\x08\x2a\x86\x48\x86\xf7\x0d\x02\x05\x05\x00\x04\x10',
@@ -44,7 +49,7 @@ HASH_ASN1 = {
     'SHA-512': b'\x30\x51\x30\x0d\x06\x09\x60\x86\x48\x01\x65\x03\x04\x02\x03\x05\x00\x04\x40',
 }
 
-HASH_METHODS = {
+HASH_METHODS: typing.Dict[str, typing.Callable[[], HashType]] = {
     'MD5': hashlib.md5,
     'SHA-1': hashlib.sha1,
     'SHA-224': hashlib.sha224,

--- a/tests/test_mypy.py
+++ b/tests/test_mypy.py
@@ -1,4 +1,5 @@
 import pathlib
+import sys
 import unittest
 
 import mypy.api
@@ -9,8 +10,11 @@ test_modules = ['rsa', 'tests']
 class MypyRunnerTest(unittest.TestCase):
     def test_run_mypy(self):
         proj_root = pathlib.Path(__file__).parent.parent
-        args = ['--incremental', '--ignore-missing-imports'] + [str(proj_root / dirname) for dirname
-                                                                in test_modules]
+        args = [
+            '--incremental',
+            '--ignore-missing-imports',
+            f'--python-version={sys.version_info.major}.{sys.version_info.minor}'
+        ] + [str(proj_root / dirname) for dirname in test_modules]
 
         result = mypy.api.run(args)
 

